commit 4e66d0f7ca6ef2cac6c303e761af605c82b78456
Author: petarj <petarj@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Tue Jan 31 18:22:20 2017 +0000

    force old implementation of std::string for leak_cpp_interior test
    
    This patch forces leak_cpp_interior to be compiled using old implementation
    of std::string.
    
    Related issue #373069
    
    Patch by Aleksandar Rikalo.
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@16217 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/memcheck/tests/leak_cpp_interior.cpp b/memcheck/tests/leak_cpp_interior.cpp
index f66fc89..a4b4824 100644
--- a/memcheck/tests/leak_cpp_interior.cpp
+++ b/memcheck/tests/leak_cpp_interior.cpp
@@ -1,3 +1,4 @@
+#define _GLIBCXX_USE_CXX11_ABI 0
 #define __STDC_FORMAT_MACROS
 #include <inttypes.h>
 #include <stdio.h>
diff --git a/memcheck/tests/leak_cpp_interior.stderr.exp b/memcheck/tests/leak_cpp_interior.stderr.exp
index 3228f74..70e2764 100644
--- a/memcheck/tests/leak_cpp_interior.stderr.exp
+++ b/memcheck/tests/leak_cpp_interior.stderr.exp
@@ -2,8 +2,8 @@
 valgrind output will go to log
 VALGRIND_DO_LEAK_CHECK
 4 bytes in 1 blocks are definitely lost in loss record ... of ...
-   by 0x........: doit() (leak_cpp_interior.cpp:115)
-   by 0x........: main (leak_cpp_interior.cpp:130)
+   by 0x........: doit() (leak_cpp_interior.cpp:116)
+   by 0x........: main (leak_cpp_interior.cpp:131)
 
 LEAK SUMMARY:
    definitely lost: 4 bytes in 1 blocks
diff --git a/memcheck/tests/leak_cpp_interior.stderr.exp-64bit b/memcheck/tests/leak_cpp_interior.stderr.exp-64bit
index 7a862a3..612fa3e 100644
--- a/memcheck/tests/leak_cpp_interior.stderr.exp-64bit
+++ b/memcheck/tests/leak_cpp_interior.stderr.exp-64bit
@@ -2,8 +2,8 @@
 valgrind output will go to log
 VALGRIND_DO_LEAK_CHECK
 8 bytes in 1 blocks are definitely lost in loss record ... of ...
-   by 0x........: doit() (leak_cpp_interior.cpp:115)
-   by 0x........: main (leak_cpp_interior.cpp:130)
+   by 0x........: doit() (leak_cpp_interior.cpp:116)
+   by 0x........: main (leak_cpp_interior.cpp:131)
 
 LEAK SUMMARY:
    definitely lost: 8 bytes in 1 blocks
diff --git a/memcheck/tests/leak_cpp_interior.stderr.exp-64bit-solaris b/memcheck/tests/leak_cpp_interior.stderr.exp-64bit-solaris
index d53e4ce..f7e1a07 100644
--- a/memcheck/tests/leak_cpp_interior.stderr.exp-64bit-solaris
+++ b/memcheck/tests/leak_cpp_interior.stderr.exp-64bit-solaris
@@ -2,8 +2,8 @@
 valgrind output will go to log
 VALGRIND_DO_LEAK_CHECK
 8 bytes in 1 blocks are definitely lost in loss record ... of ...
-   by 0x........: doit() (leak_cpp_interior.cpp:115)
-   by 0x........: main (leak_cpp_interior.cpp:130)
+   by 0x........: doit() (leak_cpp_interior.cpp:116)
+   by 0x........: main (leak_cpp_interior.cpp:131)
 
 LEAK SUMMARY:
    definitely lost: 8 bytes in 1 blocks
diff --git a/memcheck/tests/leak_cpp_interior.stderr.exp-solaris b/memcheck/tests/leak_cpp_interior.stderr.exp-solaris
index 80cd5c3..f9fc390 100644
--- a/memcheck/tests/leak_cpp_interior.stderr.exp-solaris
+++ b/memcheck/tests/leak_cpp_interior.stderr.exp-solaris
@@ -2,8 +2,8 @@
 valgrind output will go to log
 VALGRIND_DO_LEAK_CHECK
 4 bytes in 1 blocks are definitely lost in loss record ... of ...
-   by 0x........: doit() (leak_cpp_interior.cpp:115)
-   by 0x........: main (leak_cpp_interior.cpp:130)
+   by 0x........: doit() (leak_cpp_interior.cpp:116)
+   by 0x........: main (leak_cpp_interior.cpp:131)
 
 LEAK SUMMARY:
    definitely lost: 4 bytes in 1 blocks
